java/tools/RunDFAMain.kt

```diff
@@ -0,0 +1,126 @@
+package arcs.core.analysis
+
+import arcs.core.data.Check
+import arcs.core.data.proto.ManifestProto
+import arcs.core.data.proto.decodeRecipes
+import java.io.File
+import java.util.concurrent.TimeUnit
+
+fun main(args: Array<String>) {
+    if (args.size == 0) {
+        print(
+            """
+              Usage:
+                SIGH_CMD=/path/to/sigh run_dfa <file1> <file2> ...
+            """.trimIndent()
+        )
+        return
+    }
+    val sighCmd: String? = System.getenv("SIGH_CMD")
+    if (sighCmd == null) {
+        print("Set `SIGH_CMD` environment variable to the sigh command binary.")
+        return
+    }
+    val dfaRunner = DFARunner(sighCmd)
+    args.forEach { dfaRunner.verifyChecksInFile(it) }
+}
+
+class DFARunner(val sighCmd: String) {
+    private fun Check.asString(): String {
+        this as Check.Assert
+        return "${accessPath} is $predicate"
+    }
+
+    fun manifestToProto(manifestFile: String): ManifestProto {
+        val WORKING_DIR = System.getProperty("user.dir")
+        // File("/usr/local/google/home/bgogul/workspace/cerebra/arcs")
+        val manifestFileAbsolutePath = if (manifestFile.startsWith('/')) {
+            manifestFile
+        } else {
+            "$WORKING_DIR/$manifestFile"
+        }
+        val tmpDir = System.getProperty("java.io.tmpdir")
+        val tmpFile = File
+            .createTempFile("tmp", ".pb.bin", File(tmpDir))
+            .apply { deleteOnExit() }
+        val tmpFilePath = tmpFile.getAbsolutePath()
+        val sighCommandArgs = listOf(
+            sighCmd,
+            "manifest2proto",
+            manifestFileAbsolutePath,
+            "--outfile",
+            tmpFile.getName(),
+            "--outdir",
+            tmpFile.getParent()
+        )
+        ProcessBuilder(sighCommandArgs)
+            .directory(File(WORKING_DIR))
+            .redirectOutput(ProcessBuilder.Redirect.INHERIT)
+            .redirectError(ProcessBuilder.Redirect.INHERIT)
+            .start()
+            .waitFor(60, TimeUnit.MINUTES)
+        return ManifestProto.parseFrom(File(tmpFilePath).readBytes())
+    }
+
+    fun verifyChecksInFile(manifestFile: String) {
+        val INGRESS_PREFIX = "// #Ingress:"
```

Comment:
Is any of this shareable with the test utils? It might be good to pull them out into something reusable.

Reply:
Parsing of the manifest for ingresses and expected violations can be refactored out. I will do so in a separate PR.